docs/Guides/Recommendations.md

```diff
@@ -349,3 +350,29 @@ It is perfectly fine to spin up several Fastify instances within the same
 Node.js process and run them concurrently, even in high load systems. 
 Each Fastify instance only generates as much load as the traffic it receives,
 plus the memory used for that Fastify instance.
+
+## Causes Of Performance Degradation
+<a id="performance"></a>
+
+### Routes
+
+Regular expression routes are supported, however RegExp is very expensive in
+terms of performance.
+```js
+// parametric with regexp
+fastify.get('/example/:file(^\\d+).png', function (request, reply) {
```

Comment:
It is actually wrong, parametric path always be RegExp for `find-my-way` in `fastify@4`.

Reply:
Did we run some benchmarks about it in the past?

I think `found-my-way` ones does not includes such complex regex